zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreterServer.java

```diff
@@ -142,6 +141,7 @@
   private ScheduledExecutorService resultCleanService = Executors.newSingleThreadScheduledExecutor();
 
   private boolean isTest;
+  private boolean isFlinkK8sApplicationMode = false;
```

Comment:
`RemoteInterpreterServer` is a high-level object that handles `Interpreter`s identically. It seems a little odd that this object needs to know the operation mode of a particular interpreter.

Reply:
That's right, but I think this is the only way for flink k8s mode. For flink k8s mode, the `RemoteInterpreterServer#main` is the not the actual main entry point, `RemoteInterpreterServer#main` will be called by flink framework via reflection. So if we call `System.exit()` method in flink's k8s mode, it would make flink framework think that the flink job is exited abnormally, and will restart it again which is not what we expect. Let me add some comment to explain it in this class

Reply:
I agree with @cuspymd, changing `RemoteInterpreterServer` for a specific interpreter is not good.
In general, calling `System.exit()` is not good style, so we should wait for #4089 and help @PrarthiJain if necessary. This PR removes a `System.exit(0)`.

Maybe we are also able to remove the second [`System.exit(1)`](https://github.com/apache/zeppelin/blob/97ce203a7fb72fd0300ff84ea5fa8e52f6125ebc/zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreterServer.java#L705)

If we cannot remove the `System.exit()`, then we should work with an interpreter-independent variable that should be set via the start command or evaluated at start-up.

Reply:
No problem, let's wait for  #4089